### ISSUE TITLE
Separate Docker Netwoks to increase security

### ISSUE THREAD

**Issue by Rustypredator** [pterodactyl/panel#1759]
Im concerned, that the current implementation with using one server-wide network for all servers is not very secure.  
For example: in a Minecraft-Proxy Network the individual servers are exposed to the local network without authentication.  
This would allow another user to create their own Proxy and connect to the Servers and avoiding authenticating with the right Proxy and let them use offline Accounts.  

The simplest solution (i think) would be to create one network per server and per user so their own servers can see each other but not the servers of the other users  

The possibility that one user finds out what exactly the ip and port of another users server is are pretty slim, but it still exists...

**Comment by KermanIsPretty** [comment]
If your a hosting company why not just create a pterodactyl instance in a container in the first place giving them full admin rights to make servers but capping their container on what they paid for?

**Comment by DaneEveritt** [comment]
Yeah, the best solution for this is per-user networks, it just gets slightly complicated on the Daemon side since it'll need to properly group the servers together, that way a user with all of their servers on one machine can have them networked together properly.

Do you know if other hosting software avoids this as well, or are they also all vulnerable to the same issue (I'd imagine so, especially in non-docker environments). 

This also adds some nice complexity to the configuration for bungee networks in the configuration files since the daemon will need to be able to determine the correct networking to use for each configuration file as `127.0.0.1` obviously wouldn't work.

**Comment by Rustypredator** [comment]
i dont know about other hosting software, since they are mostly not free 😄   
bungee config works very well with hostnames, i use the container name + port in the config and never had any problems

**Comment by DaneEveritt** [comment]
Even using host names wouldn't solve this issue anyways, its the same underlying networking functionality.

**Comment by PurpleIsEverything** [comment]
Should probably add, Pterodactyl's current Daemon already offers support for this just takes a bit extra setup.

A while back I added support for the weavemesh networking type a swarm-less multi-host docker network solution. This solution also provides container isolation as outlined [HERE](https://www.weave.works/docs/net/latest/tasks/manage/application-isolation/)

There is nothing preventing one from using this network with a single host and it's fairly easy to setup as well as it runs in Docker too.

**Comment by LeCodeCo** [comment]
@PurpleIsEverything Me and @matthewpi got weavemesh working normally but we ran into a slight issue for separating internal networks with it.  The issue lies with how to handle handling giving internal subnets, as, from my understanding of the docs, you just add a container to an isolated IP subnet such as /24. If we gave each user a /24, we'd be restricting each account to have 254 servers.

I was wondering if you had any more experience or knowledge which could help?

**Comment by schrej** [comment]
Hostnames actually would resolve this I think. As both the bungee as well as the servers are in the same network the containers can communicate anyway. You (not wings?) would just need to figure out the container IPs of the servers you want to attach. When containers are rebuilt, those can change though. As hostnames are properly resolved within docker networks, you can just refer to the other servers with their hostname (maybe just the short uuid? Or show it in the panel? Or generate it from the name?) and don't have to care about what subnet you are in.

**Comment by tim-van-dijkhuizen** [comment]
I like the idea of having a separate network for each server by default. The user-wide network might be a bit confusing when you're working with multiple nodes. Instead, I would like an [option for admins to create custom networks and assign servers to it](https://github.com/pterodactyl/panel/issues/3545). This way, you have to think about what you're doing instead of expecting everything to be in the same network.

**Comment by DaneEveritt** [comment]
In the interest of maintainer sanity and cleaning up an enormous backlog of stale feature requests we have moved to requiring all feature requests be discussed within [Discussions](https://github.com/orgs/pterodactyl/discussions/categories/feature-requests). Please consider re-opening this request there if you believe it is still relevant and worth consideration.

See https://github.com/pterodactyl/panel/issues/5491 for additional context.

Thanks.

<sub>_This comment is in the interest of issue tracker maintenance: it does not imply that maintainers are endorsing or dismissing your request._</sub>
